docs/template-engine.md

```diff
@@ -0,0 +1,152 @@
+# Template Engine 
+
+OTel Weaver is capable of generating documentation or code from a semantic
```

Comment:
Before we spend more time going down this path, we should validate this is desired from the rest of SEmconv maintainers.


There are several types of codegen I think we'll want going forward:


1. We have human-written instructions for semantic conventions where-in we *embed* rendered markdown outlining the state of the semconv registry, via `<-- semconv( render command) -->`.  I'd prefer if we worked towards feature parity on this behavior first.
2. We have basic rendering of the "registry itself" as a generic "{language}doc" like output.  While I don't think that's exactly what you had in mind in this PR, it's the second thing I'd like to see.
3. We will need to generate "{langauge}doc" like output for "application" instrumentation.  I believe this is what you're creating here, and I really like the direction, but think we should first make sure 1+2 are solved.
4. We will need to generate code.

I actually think we should keep these four use cases as *separate* things for weaver.

TL;DR; I think this currently makes another good demo, but we need to start socializing it so it can go further.   Additionally, let's see if we can replicate some key use cases of semconv generation today.